dotCMS/src/test/java/com/dotcms/filters/NormalizationFilterTest.java

```diff
@@ -111,5 +135,73 @@ private void compare(final String originalURI, final String expectedNormalizedUR
         assertNotNull(normalizedValueByFilter);
         assertEquals(expectedNormalizedURI, normalizedValueByFilter);
     }
+    
+    /**
+     * Test to verify the {@link NormalizationFilter} is applying properly the normalization on
+     * invalids URIs
+     */
+    @Test
+    public void test_uri_normalizer_fixes_double_slashes() throws IOException, ServletException {
+
+        
+        // remove all //, replace with /
+        String originalURI = "//folder/important/secret_file.dat";
+        String expectedNormalizedURI = "/folder/important/secret_file.dat";
+        compare(originalURI, expectedNormalizedURI);
+
+        // testing ///
+        originalURI = "///folder/important/secret_file.dat";
+        compare(originalURI, expectedNormalizedURI);
+
+        // testing ////
+        originalURI = "////folder/important/secret_file.dat";
+        compare(originalURI, expectedNormalizedURI);
+        
+
+        
+        // testing multiple ////
+        originalURI = "////folder/important///secret_file.dat";
+        compare(originalURI, expectedNormalizedURI);
+        
+        // testing // not at the root (query strings are not part of URI)
+        originalURI = "/test//folder/folder1//forward_jsp.jsp?FORWARD_URL=http://google.com";
+        expectedNormalizedURI = "/test/folder/folder1/forward_jsp.jsp";
+        compare(originalURI, expectedNormalizedURI);
+
+        // A ".." segment is removed only if it is preceded by a non-".." segment
+        originalURI = "///test/../folder/important//../secret_file.dat";
+        expectedNormalizedURI = "/folder/secret_file.dat";
+        compare(originalURI, expectedNormalizedURI);
+
+        // Each "." segment is simply removed
+        originalURI = "./f/older//folder1//file.dat";
+        expectedNormalizedURI = "/f/older/folder1/file.dat";
+        compare(originalURI, expectedNormalizedURI);
+
+        // Each "." segment is simply removed
+        originalURI = "./folder/./folder1/file.dat//..//";
+        expectedNormalizedURI = "/folder/folder1/";
+        compare(originalURI, expectedNormalizedURI);
+
+        // starts with ..//
+        originalURI = "..//folder/./folder1//file.dat";
+        expectedNormalizedURI = "/folder/folder1/file.dat";
+        compare(originalURI, expectedNormalizedURI);
+    }
+    
+    
+    /**
+     * Test to verify the {@link NormalizationFilter} is applying properly the normalization on
+     * invalids URIs
+     */
+    @Test
+    public void test_uri_normalizer_invalid_uris() throws IOException, ServletException {
+        // testing escaped slashes - Filter barfs and returns /
+        String originalURI = "/\\///folder//important////secret_file.dat";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'originalURI' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5436097)